Add post for hacktoberfest 2021 (#157)
void-linux/void-linux.github.io: _posts/2021-09-23-hacktoberfes.markdown

---
title: Hacktoberfest 2021
layout: post
---

> Are you ready for Hacktoberfest 2021?  Void Linux is!  This year we're

Reviewer: ```suggestion
Are you ready for Hacktoberfest 2021?  Void Linux is!  We're
```

> excited to be participating for our //// year.  We're looking for

Reviewer 2: 5th

> excited to be participating for our //// year.  We're looking for
> contributions to assist this year with our out-of-date packages queue,
> and these are all great ways to dip your feet into the world of Linux

Reviewer: ```suggestion
excited to be participating for our //// year.  Contributions that help to
address our out-of-date packages queue are especially welcome.
These are all great ways to dip your feet into the world of Linux
```

> distro package management, and what happens behind the scenes to make
> sure your system is up to date and complete with a wide selection of
> packages.

Reviewer: ```suggestion
distro package management and what happens behind the scenes to 
provide a wide selection of packages and make sure your system remains up to date.
```

Updating packages is very easy.  You can select a package from the
list of [out of date
packages](http://alpha.de.repo.voidlinux.org/void-updates/void-updates.txt)
and update it using the tools in the [void-packages
repo](https://github.com/void-linux/void-packages).  The
[manual](https://github.com/void-linux/void-packages/blob/master/Manual.md)
might be of assistance when you are updating packages.

As a general rule, we would prefer Hacktoberfest contributors that do
not have a previous track record with the Void Linux project steer
clear of "structural" packages unless you have specific domain

> knowledge that qualifies you to work on high-risk packages.  When
> selecting a package to update, select a package registered to
> `orphan@voidlinux.org` when possible.  These packages are otherwise
> unmaintained, and your contribution will have a bigger impact.  You
> can update packages that are owned by existing developers, but
> understand that conflicts between a maintainer and contributor will be
> resolved at the discretion of Void staff.

Reviewer 3: ```suggestion
knowledge that qualifies you to work on high-risk packages.  When
selecting a package to update, prefer packages registered to
`orphan@voidlinux.org`.  These packages are otherwise
unmaintained, and your contribution will have a bigger impact.  You
can update packages that have a maintainer assigned, but
understand that conflicts between a maintainer and contributor will be
resolved at the discretion of Void staff.
```
This section could be interpreted as "please don't touch packages which are not orphaned" which is not the intention. The changes I suggested make it a bit better, but the last part still sounds a bit too hostile in my opinion: 

> but understand that conflicts between a maintainer and contributor will be
resolved at the discretion of Void staff.

Here are some useful tips when updating packages:

  * While we're not completely opposed to PRs that add new packages,
    you're much more likely to get your PR approved and merged if its
    a well written update.

>   * Don't PR broken code, our maintainers are much less likely to give

Reviewer: ```suggestion
  * Don't PR broken code. Our maintainers are much less likely to give
```

>     a second look to a PR that didn't build when it was submitted.

Reviewer 3: This is a bit tricky, tests could be seen as broken builds. I sometimes see that some don't know how to cross compile or run tests on their machine, so they rely on CI. We could link the relevant parts of documentation.

Author: I think this is as good as we're going to get.  We put the "I've built this and it works" checkbox right in PRs and people ignore that, adding it in more places isn't really going to change whether people pay attention or not.

  * While it's possible to run xbps-src from an alien distro, this
    isn't really supported.  If you're a seasonsed Linux user and want
    to try Void, now is the time!

>   * The update list is sometimes wrong, we'd love to get patches that

Reviewer: ```suggestion
  * The update list is sometimes wrong. We'd love to get patches that
```

    improve its reliability by ignoring beta versions or adding checks

>     to packages that are not detected as out of date correctly.

Reviewer 3: we could link the documentation about the update file here

>     to packages that are not detected as out of date correctly.

Reviewer: ```suggestion
    to packages that are not correctly detected as out of date.
```

  * If you have expertise in C, GNU Autotools, or other build systems,

>     taking a look at projects that we've marked as not
>     cross-compilation compatible and fixing the upstream issue can be

Reviewer: ```suggestion
    taking a look at projects that we've marked as incompatible
    with cross compliation and fixing the upstream issue can be
```

    an amazing contribution that impacts more than just Void.

We look forward to working with the amazing world of open source
developers this month to improve Void and continue our high standards
for quality and reliability.  To ensure your PR has the best chance at
being accepted, feel free to reach out for help as explained in the

> manual.  Together this will be a high impact Hacktoberfest.

Reviewer: ```suggestion
manual.  Together, we can make this a high-impact Hacktoberfest.
```
